libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs

```diff
@@ -61,6 +64,78 @@ public async Task<HttpResponseMessage> CallAPIAsync(string path, FacebookMessage
             }
         }
 
+        /// <summary>
+        /// Get a Facebook API client with the correct credentials based on the page identified in the incoming activity.
+        /// This is used by many internal functions to get access to the Facebook API, and is exposed as `bot.api` on any BotWorker instances passed into Botkit handler functions.
+        /// </summary>
+        /// <param name="activity">An incoming message activity.</param>
+        /// <returns>A Facebook API client.</returns>
+        public async Task<FacebookClientWrapper> GetAPIAsync(Activity activity)
+        {
+            if (!string.IsNullOrWhiteSpace(Options.AccessToken))
+            {
+                return new FacebookClientWrapper(new FacebookAdapterOptions(Options.VerifyToken, Options.AppSecret, Options.AccessToken));
+            }
+            else
+            {
+                if (!string.IsNullOrWhiteSpace(activity.Recipient?.Id))
```

Comment:
This if can be `if (string.IsNullOrWhiteSpace(activity.Recipient?.Id))` throw the exception removing the last else statement.